core/translate/upsert.rs

```diff
@@ -525,35 +523,35 @@ fn rewrite_upsert_expr_in_place(
     conflict_rowid_reg: usize,
     insertion: &Insertion,
 ) -> crate::Result<()> {
-    use ast::Expr;
+    use ast::{Expr, Name};
 
-    // helper: return the CURRENT-row register for a column (including rowid alias)
     let col_reg = |name: &str| -> Option<usize> {
         if name.eq_ignore_ascii_case("rowid") {
             return Some(conflict_rowid_reg);
         }
-        let (idx, _c) = table.get_column_by_name(&normalize_ident(name))?;
+        let (idx, _) = table.get_column_by_name(&normalize_ident(name))?;
         Some(current_start + idx)
     };
+
     walk_expr_mut(e, &mut |expr: &mut ast::Expr| -> crate::Result<()> {
         match expr {
-            // EXCLUDED.x -> insertion register
-            Expr::Qualified(ns, ast::Name::Ident(c)) => {
-                if ns.as_str().eq_ignore_ascii_case("excluded") {
-                    let Some(reg) = insertion.get_col_mapping_by_name(c.as_str()) else {
+            // EXCLUDED.x or t.x (t may be quoted)
+            Expr::Qualified(ns, Name::Ident(c) | Name::Quoted(c))
+            | Expr::DoublyQualified(_, ns, Name::Ident(c) | Name::Quoted(c)) => {
+                if is_ns(ns, "excluded") {
```

Comment:
normalize ident before this call so we don't do it twice? then we could just do use `eq_ignore_ascii_case` without the `is_ns` function which currently only has one usage